From babea3fd61275d479e91ef58c85728054f368289 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 13 Apr 2016 23:16:17 -0400
Subject: [PATCH 06/62] geoloc.py: support https

...
13:04:28,029 DEBUG anaconda: Geoloc: URLError for Fedora GeoIP API lookup:
<urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)>
...

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/geoloc.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/geoloc.py b/pyanaconda/geoloc.py
index 7b8dc0d..e7e31e6 100644
--- a/pyanaconda/geoloc.py
+++ b/pyanaconda/geoloc.py
@@ -102,6 +102,7 @@ in a couple seconds
 
 import urllib
 import urllib2
+import ssl
 import json
 import dbus
 import threading
@@ -525,8 +526,11 @@ class FedoraGeoIPProvider(GeolocationBackend):
 
     def _refresh(self):
         try:
+            context = ssl._create_unverified_context()
+            log.debug("context %s" % context)
             reply = urllib2.urlopen(self.API_URL, timeout=
-                                    constants.NETWORK_CONNECTION_TIMEOUT)
+                                    constants.NETWORK_CONNECTION_TIMEOUT,
+                                    context=context)
             if reply:
                 json_reply = json.load(reply)
                 territory = json_reply.get("country_code", None)
-- 
2.7.4

